include/pagmo/algorithms/pso.hpp

```diff
@@ -0,0 +1,875 @@
+                /*  ''Clerc's analysis of the iterative system led him to propose a strategy for the
+                 *  placement of "constriction coefficients" on the terms of the formulas; these
+                 *  coefficients controlled the convergence of the particle and allowed an elegant and
+                 *  well-explained method for preventing explosion, ensuring convergence, and
+                 *  eliminating the arbitrary Vmax parameter. The analysis also takes the guesswork
+                 *  out of setting the values of phi_1 and phi_2.''
+                 *  ''this is the canonical particle swarm algorithm of today.''
+                 *  [Poli et al., 2007] http://dx.doi.org/10.1007/s11721-007-0002-0
+                 *  [Clerc and Kennedy, 2002] http://dx.doi.org/10.1109/4235.985692
+                 *
+                 *  This being the canonical PSO of today, this variant is set as the default in PaGMO.
+                 *-------------------------------------------------------------------------------------*/
+                else if (m_variant == 5u) {
+                    for (decltype(dim) d = 0u; d < dim; ++d) {
+                        r1 = drng(m_e);
+                        r2 = drng(m_e);
+                        m_V[p][d] = m_omega * (m_V[p][d] + m_eta1 * r1 * (lbX[p][d] - X[p][d])
+                                               + m_eta2 * r2 * (best_neighb[d] - X[p][d]));
+                    }
+                }
+
+                /*-------Fully Informed Particle Swarm-------------------------------------------------*/
+                /*  ''Whereas in the traditional algorithm each particle is affected by its own
+                 *  previous performance and the single best success found in its neighborhood, in
+                 *  Mendes' fully informed particle swarm (FIPS), the particle is affected by all its
+                 *  neighbors, sometimes with no influence from its own previous success.''
+                 *  ''With good parameters, FIPS appears to find better solutions in fewer iterations
+                 *  than the canonical algorithm, but it is much more dependent on the population topology.''
+                 *  [Poli et al., 2007] http://dx.doi.org/10.1007/s11721-007-0002-0
+                 *  [Mendes et al., 2004] http://dx.doi.org/10.1109/TEVC.2004.826074
+                 *-------------------------------------------------------------------------------------*/
+                else if (m_variant == 6u) {
+                    for (decltype(dim) d = 0u; d < dim; ++d) {
+                        sum_forces = 0.;
+                        for (decltype(neighb[p].size()) n = 0u; n < neighb[p].size(); ++n) {
+                            sum_forces += drng(m_e) * acceleration_coefficient * (lbX[neighb[p][n]][d] - X[p][d]);
+                        }
+                        m_V[p][d] = m_omega * (m_V[p][d] + sum_forces / static_cast<double>(neighb[p].size()));
```

Comment:
This ``static_cast`` is superfluous, as you are using it as a divisor with a ``double`` dividend: the compiler will take care of do the casting for you.